### DISCUSSION TITLE
How do I find the Methods and Properties for ESPAsyncWebServer by Lacamera ver 3.1.0?

### DISCUSSION THREAD

**Discussion by fester225** [community/community#114309] [Programming Help]
### Select Topic Area

Question

### Body

It looks as though ESPAsyncWebServer (Lacamera version 3.1.0) is going to be very helpful to me. In the meantime, I haven't been able to find a source for its Methods and Properties. It seems as though this kind of information should be available from a standardized source. So far I haven't seen one. There also seems to be the problem of libraries which depend on lower level libraries to operate. It isn't always obvious what libraries they depend on, let alone what Methods and Properties they use but the upper level library doesn't tell you about. Where do I find this?

**Comment by monmoymondol** [comment]
Hey fester225,

You’re right — documentation for the Lacamera version 3.1.0 of ESPAsyncWebServer isn’t easy to find. The best way to see all available methods and properties is to look directly at the source code and header files.

Try this:
• Go to the GitHub repo for ESPAsyncWebServer (Lacamera) and open the ESPAsyncWebServer.h and ESPAsyncWebServer.cpp files — they list all classes, functions, and properties.
• You can also check the examples folder in the repo; it shows how most methods are used in real sketches.
• If you’re using PlatformIO or Arduino IDE, hover over a function name and press Ctrl + Click (or Cmd + Click on Mac) — it jumps straight to the method definition.

There isn’t an official API reference yet, but reading the header files gives you everything you need. Once you get familiar with the class structure, you’ll see how it connects to lower‑level libraries like AsyncTCP.

Hope that helps — it’s a solid library once you get used to exploring the source!
